.circleci/config.yml

```diff
@@ -47,89 +44,98 @@ jobs:
       - restore_cache: *restore_cargo_cache
 
       - run:
-          name: Check for Errors
-          command: |
-            cargo clean
-            cargo check
-            cd fixture_setup
-            cargo check
+          name: Check Formatting of forte_core
+          command: *lint_commands
 
-      - save_cache: *save_cargo_cache
+      - run:
+          name: Check Formatting of fixture_setup
+          working_directory: fixture_setup
+          command: *lint_commands
 
       - run:
-          name: Check Formatting
-          command: |
-            cargo fmt -- --write-mode=diff
-            cd fixture_setup
-            cargo fmt -- --write-mode=diff
+          name: Compile forte_core
+          command: *compile_commands
 
       - run:
-          name: Check Linting
-          command: |
-            cargo clippy -- -D warnings
-            cd fixture_setup
-            cargo clippy -- -D warnings
+          name: Compile fixture_setup
+          working_directory: fixture_setup
+          command: *compile_commands
 
       - save_cache: *save_cargo_cache
 
-  test:
+      # The workspace stores some of the same information the cache does, but
+      # the workflow is shared across a workflow and updated on every build
+      # while the cache only updates when a lock file changes.
+      - persist_to_workspace:
+          root: .
+          paths:
+            - target
+            - fixture_setup/target
+
+  integration-test:
     docker:
-      # Primary container
       - image: fortemusic/core-build
 
-      # Secondary container (available on localhost)
-      - image: redis:4.0.6
+    environment:
+      RUST_BACKTRACE: "1"
 
     steps:
+      - attach_workspace:
+          at: /project
+
       - checkout
 
       - run:
           name: Checkout Git Submodules
           command: git submodule update --init
 
-      - restore_cache: *restore_cargo_cache
-
-      - restore_cache: *restore_fixture_cache
-
       - restore_cache: *restore_yarn_cache
 
       - run:
-          name: Install Schema
+          name: Install Test Runner
           working_directory: fixture_setup
-          command: yarn install
+          command: yarn
 
-      - run:
-          name: Install Fixtures
-          working_directory: fixture_setup
-          command: cargo run
+      - save_cache: *save_yarn_cache
 
-      - save_cache: *save_fixture_cache
+      - run:
+          name: Initialize Database
+          environment:
+            DATABASE_URL: ./db.sqlite
+          command: diesel migration run
 
-      - save_cache: *save_yarn_cache
+      - restore_cache: *restore_cargo_cache
```

Comment:
If this also caches the same thing as the workspace, then the workspace changes might get overridden by the cache.

Reply:
It might. Checking is really hard though. Should we make the things held by the cache disjoint from what is kept in the work space? Will removing target from the cache slow down compilation?

Reply:
Just restore the workspace changes after the general cache.

Reply:
Ok, that's a good way to go about it.

Reply:
That's a good reason.